apps/condo/domains/meter/schema/MeterReadingExportTask.test.js

```diff
@@ -0,0 +1,521 @@
+
+            await expectToThrowGQLError(async () => {
+                await updateTestMeterReadingExportTask(userClient, task.id, { status: CANCELLED })
+            }, ERRORS.STATUS_IS_ALREADY_COMPLETED)
+        })
+    })
+})
+
+describe('exportMeterReadings', () => {
+    it('should create `MeterReadingExportTask` and create xlsx file', async () => {
+        const { organization, userClient } = await makeAdminClientWithRegisteredOrganizationWithRoleWithEmployee({
+            canManageProperties: true,
+            canManageMeters: true,
+            canManageMeterReadings: true,
+            canReadMeters: true,
+        })
+        const [property] = await createTestProperty(userClient, organization)
+        
+        const meterReadings = []
+
+        const [resource] = await MeterResource.getAll(userClient, { id: COLD_WATER_METER_RESOURCE_ID })
+        const [source] = await MeterReadingSource.getAll(userClient, { id: CALL_METER_READING_SOURCE_ID })
+
+        const metersCount = 10
+        const meterReadingsOnEachMeter = 10
+        const meterReadingsCount = meterReadingsOnEachMeter * metersCount
+
+        for (let i = 0; i < metersCount; i++) {
+            const [meter] = await createTestMeter(userClient, organization, property, resource, {})
+
+            for (let j = 0; j < meterReadingsOnEachMeter; j++) {
+                const [meterReading] = await createTestMeterReading(userClient, meter, source)
+
+                meterReadings.push(meterReading)
+            }
+        }
+
+        const [task] = await createTestMeterReadingExportTask(userClient, userClient.user, {
+            where: {
+                organization: {
+                    id: organization.id,
+                },
+            },
+            sortBy: 'createdAt_ASC',
+            locale: 'ru',
+            timeZone: 'Europe/Moscow',
+        })
+
+        const { timeZone, locale } = task
+
+        await waitFor(async () => {
+            const updatedTask = await MeterReadingExportTask.getOne(userClient, { id: task.id })
+
+            expect(updatedTask.file).toBeDefined()
+            expect(updatedTask.file.publicUrl.length).toBeGreaterThan(1)
+            expect(updatedTask).toHaveProperty('exportedRecordsCount', meterReadingsCount)
+            expect(updatedTask).toHaveProperty('totalRecordsCount', meterReadingsCount)
+            expect(updatedTask.exportedRecordsCount).toEqual(meterReadings.length)
+            expect(updatedTask).toHaveProperty('v', 3)
+            expect(updatedTask).toHaveProperty('status', COMPLETED)
+        })
+
+        const updatedTask = await MeterReadingExportTask.getOne(userClient, { id: task.id })
+        const url = updatedTask.file.publicUrl.replace(conf.SERVER_URL, userClient.serverUrl)
+        const filename = getTmpFile('xlsx')
+        await downloadFile(url, filename)
+        const data = await readXlsx(filename)
+
+        expectDataFormat(data,  [
+            [
+                'Дата снятия',
```

Comment:
Is it possible to test this in `en` locale, so it can be understood by third-party contributors?